bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SharedComponentPropertiesBindingsValuesProvider.java

```diff
@@ -0,0 +1,108 @@
+package com.adobe.acs.commons.wcm.impl;
+
+import com.adobe.acs.commons.wcm.PageRootProvider;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.components.Component;
+import com.day.cq.wcm.commons.WCMUtils;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.scripting.api.BindingsValuesProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.script.Bindings;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Bindings Values Provider that adds bindings for globalProperties,
+ * sharedProperties, and mergedProperties maps.
+ *
+ * globalProperties contains the shared properties accessible by
+ * all components.
+ *
+ * sharedProperties contains the shared properties specific to the
+ * current component.
+ *
+ * mergedProperties is a merge of the instance-level, shared, and
+ * global properties for the current component, giving preference
+ * to instance-level values, then shared values, and finally global
+ * values when properties exist at multiple levels with the same name.
+ */
+@org.apache.felix.scr.annotations.Component(
```

Comment:
This does not require a configuration, but only because it depends on your PageRootProvider to be satisfied.